src/sql/src/plan/statement.rs

```diff
@@ -764,11 +765,15 @@ impl<'a> StatementContext<'a> {
         // representation on `pgrepr::Type` promises to
         // produce an unqualified type name that does
         // not require quoting.
-        //
+        let mut ty = if ty.oid() < FIRST_MATERIALIZE_OID {
+            format!("pg_catalog.{}", ty)
+        } else {
+            // This relies on all non-PG types existing in `mz_catalog`, which is annoying.
+            format!("mz_catalog.{}", ty)
+        };
```

Comment:
Just to be safe, we should probably also add a case for `ty.oid() >= FIRST_USER_OID` and return an error.

Reply:
Done